src/app/scripts/cac/user/cac-user-preferences.js

```diff
@@ -1,67 +1,75 @@
-CAC.User.Preferences = (function(Storages, _) {
+CAC.User.Preferences = (function(_) {
     'use strict';
 
-    // TODO: figure out local storage strategy
-    // set up local storage
-    // var namespaceStorage = Storages.initNamespaceStorage('cac_otp');
-    // var storage = namespaceStorage.localStorage;
 
     // Initialize preference storage object.
-    // Currently it just uses an 'options' dictionary, so preferences lives only as long as
-    // the page for which this is initialized.
+    // All values in storages should be stringified first.
+    // Setting to a falsy value will remove the object from storages;
```

Comment:
Unintended falsiness is such a troublesome source of bugs that I'm trying to think of whether this could be a little tighter without becoming annoying.  It doesn't seem impossible that there could come a time when a zero should be treated as a zero.  I guess treating empty string as "unset" makes sense since it could be set that way by a URL param.  What about `Setting to null, undefined, or empty string will remove the object from storages`? Then the condition below would be:
```
if (!_.isUndefined(val) && !_.isNull(val) && val !== 'undefined' && val !== '') {
```
It's wordier, but not terrible.
Or flip it and have the `delete` be the conditional case, with
```
if (_.isUndefined(val) || _.isNull(val) || val === 'undefined' || val === '') {
```
That's slightly easier to read and would also short-circuit, for what that's worth.